### PR TITLE
fix(import): user permissions get cleaned up from collections

### PR DESCRIPTION
**Pull request by RomanJRW**
relates to OPT-7106

## Description

We are allowing collection role cleanup to be disabled. From 3.11, this will be disabled by default. Otherwise, this will be enabled.

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [x] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: optimize/self-managed/optimize-deployment/migration-update/3.10-to-3.11.md [hunk 1]
```diff
@@ -23,3 +23,9 @@ Optimize now supports Elasticsearch `8.7`.
 See the [supported environments]($docs$/reference/supported-environments) section for the full range of supported versions.
 
 If you need to update your Elasticsearch cluster, refer to the general [Elasticsearch update guide](https://www.elastic.co/guide/en/elasticsearch/reference/current/setup-upgrade.html). Usually, the only thing you need to do is perform a [rolling update](https://www.elastic.co/guide/en/elasticsearch/reference/current/rolling-upgrades.html).
+
+## Changes in the configuration
+
+Prior to Optimize 3.11.0, Optimize has performed collection role cleanup after syncing identities with the engine. From
+Optimize 3.11.0 onwards, this is now disabled by default. It can be reenabled by setting the
+`import.identitySync.collectionRoleCleanupEnabled` property value to `true`
```
Comment (PHWaechtler) [inline optimize/self-managed/optimize-deployment/migration-update/3.10-to-3.11.md:31]: The setting is set to `false` for the pre-3.11 example config files as well in this PR - are those supposed to be set to `true`?

File: optimize/self-managed/optimize-deployment/configuration/system-configuration-platform-7.md [hunk 2]
```diff
@@ -69,6 +69,7 @@ REST API endpoint locations, timeouts, etc.
...
 | import.identitySync.includeUserMetaData                   | true          | Whether to include metaData (firstName, lastName, email) when synchronizing users. If disabled only user IDs will be shown on user search and in collection permissions.                                                                                                                                                                                                                                                                                                                                                  |
+| import.identitySync.collectionRoleCleanupEnabled          | true          | Whether collection role cleanup should be performed                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
```
Comment (PHWaechtler) [inline optimize/self-managed/optimize-deployment/configuration/system-configuration-platform-7.md:72]: I wouldn't expect users to know what the `collection role cleanup` does, could add a bit more explanation:
```suggestion
| import.identitySync.collectionRoleCleanupEnabled          | true          | Whether collection role cleanup should be performed. If disabled, users that no longer exist in the identity provider will not be automatically removed from collection permissions                                                                                                                                                                                                                                                                                                                                                                                                                                     |
```
though that then probably will lead people to assume they need this to be `true`. Up to you 🤷

**Review by PHWaechtler — APPROVED** [review]
Lgtm just two questions

File: optimize/self-managed/optimize-deployment/migration-update/3.10-to-3.11.md (hunk 1 above)
Comment (RomanJRW) [reply optimize/self-managed/optimize-deployment/migration-update/3.10-to-3.11.md:31]: good point, fixed 👍

**Comment by RomanJRW** [comment]
Yes please @christinaausley - just because we put something in the upgrade guide, and I always trust your judgement more than my own as to whether I make sense in important updates like this 🙂

**Review by christinaausley — APPROVED** [review]
No changes for this one after review 👍

**Pull request merged**